Scala client for Plotly

pbugnion opened the pull request:
Includes tutorials and a user guide valid for version 0.1 of the Scala Plotly client.

Obviously this is still early days for the client, but it's already usable (provided you like scatter or line plots).

This PR adds:

 - an overall index for the scala client documentation
 - a tutorial on scatter plots
 -  a (short) tutorial on how to make multi-axes plots
 - a very embryonic user guide.

Before this gets merged, we still need to decide how links to the Scala plotly client documentation will be included in the home page. Given that it's still early days, it doesn't make sense to have it at the same level as the more complete Python / R clients.

tarzzz commented on _posts/scala/2016-03-22-line-and-scatter.md:
```diff
@@ -0,0 +1,282 @@
+language: scala
+page_type: example_index
+has_thumbnail: true
+display_as: chart_type
```
`chart_type` is the category we want to put this example in i.e. `basic`, `statistical` etc..
for this one it is:
`display_as: basic` ..

tarzzz commented on _posts/scala/2016-03-22-line-and-scatter.md:
```diff
@@ -0,0 +1,282 @@
+
+# Scatter plots in Scala
+
+## Simple scatter plot
```
@pbugnion : we usually maintain a separate folder for each category,
so all the 'Line and Scatter' examples go in a separate directory inside `/_posts/scala/`.

Also, individual examples are maintained in their own html files, so for each category we have an `index` file, and multiple example files. You can refer to the line and scatter charts in javascript : https://github.com/plotly/documentation/tree/source-design-merge/_posts/plotly_js/line_and_scatter/

tarzzz commented on _posts/scala/2016-03-22-line-and-scatter.md:
```diff
@@ -0,0 +1,282 @@
+
+<iframe id="graph" scrolling="no" style="border:none;" seamless="seamless" src="https://plot.ly/~pbugnion/173.embed" height="525px" width="100%"></iframe>
+
+## Line and scatter plots
```
Similarly as above, these should also be in their individual example files, and a index file to keep them linked..

tarzzz commented:
@pbugnion : Looking good so far. I have added a few comments inline..

> we still need to decide how links to the Scala plotly client documentation will be included in the home page.

For now, we can add it in `others` section in the [api](https://plot.ly/api) page (along with MATLAB, and Julia)..

tarzzz commented on _posts/scala/2016-03-22-subplots.md:
```diff
@@ -0,0 +1,86 @@
+language: scala
+page_type: example_index
+has_thumbnail: true
+display_as: multiple_axes
```
subplots should be in `basic`category..

pbugnion commented:
Thanks for the feedback. Will work on integrating those changes.

pbugnion commented:
Alright -- I think I've made the changes you suggested. Happy to make any further changes you think are useful!

tarzzz commented on _posts/scala/2016-03-22-getting-started.md:
````diff
@@ -0,0 +1,37 @@
+```
+{
+    "username": "pbugnion",
+    "api_key": "qnm2ndgaca"
````
@pbugnion : You may not want to share your API Key Publicly.. You can use a simple text as `my_api_key`.

pbugnion replied:
That's not actually my API key. In my experience, it's better to use actual examples, rather than placeholders, but happy to change this.

cldougl replied:
@pbugnion you can use: username='DemoAccount', api_key='lr1c37zw81' this way it's a tiny bit more clear that it's a demo acct and should be replaced w/ the user's credentials

tarzzz commented on _posts/2016-04-13-scala-index.html:
```diff
@@ -0,0 +1,28 @@
+	<div class="--welcome-body">
+		<!--div.--wrap-inner-->
+		<div class="--title">
+			<div class="--category-img"><img src="https://plot.ly/gh-pages/documentation/static/images/scala-small.jpg" alt=""></div>
```
You can use the image from CDN instead: http://images.plot.ly/plotly-documentation/thumbnail/scala-small.jpg

tarzzz commented on _posts/2016-04-13-scala-index.html:
```diff
@@ -0,0 +1,28 @@
+			<div class="--category-img"><img src="https://plot.ly/gh-pages/documentation/static/images/scala-small.jpg" alt=""></div>
+			<div class="--body">
+				<h1>Plotly Scala Library</h1>
+				<p>{{page.description}}</consectetur>
```
`</consectetur>` seems like a stray tag..

pbugnion replied:
Thanks, well spotted -- I think my text editor got a little excited.

tarzzz commented:
@pbugnion : added a few more minor comments/corrections. 
I have uploaded the `scala-small.jpg` icon-image to Plotly's Image CDN [here](http://images.plot.ly/plotly-documentation/thumbnail/scala-small.jpg). We have started keeping all the thumbnails, and documentation related images at the images CDN, so it is not required to be kept in the repo here itself.
You can remove the image from `all_static/images`, and use the CDN one instead..


Please address the in-line comments, and it's good to be to merged.. 💃

pbugnion commented:
@tarzzz Thanks for the feedback. I have:
 - updated the links to point to the Scala thumbnail on the CDN
 - removed the thumbnail from the repo
 - removed the stray tag
 - added Scala to the list of clients when navigating between example pages (ie., if you're on, e.g., the Matlab line-and-scatter page now, you can navigate to the line-and-scatter page in Scala).

Happy to integrate any further feedback, obviously.

cldougl commented:
@pbugnion thanks for this! You can merge it in (I'd just change the credentials to the examples ones I put in the comment.) Let me know if you don't have permission or need any help :)

pbugnion commented:
Thanks -- I've updated the credentials as you suggested.
